server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -356,9 +364,35 @@ public void onNext(final InputStream request) {
                                 throw GrpcUtil.statusRuntimeException(Code.INVALID_ARGUMENT,
                                         myPrefix + "received a message with unhandled BarrageMessageType");
                         }
+                        requestHandler.handleMessage(message);
+                        return;
+                    }
+
+                    // handle the possible error cases
+                    if (!isFirstMsg) {
+                        // only the first messages is allowed to have null metadata
+                        throw GrpcUtil.statusRuntimeException(Code.INVALID_ARGUMENT,
+                                myPrefix + "failed to receive Barrage request metadata");
+                    }
+
+                    isFirstMsg = false;
+
+                    // test whether the FlightDescriptor cmd field contains a valid Barrage object
```

Comment:
For maximum compatibility, we should make the `FlightDescriptor` cmd field optional and allow nulls/empty rather than require a barrage-style message that serves no purpose.  If we want to add action commands later, we can test for that but allowing nulls is the most user-friendly.

Reply:
After discussion, decided to enforce `FlightDescriptor` cmd field being equal to `BarrageUtil.FLATBUFFER_MAGIC` which is '0x6E687064' (the numerical representation of the ASCII "dphn".)